packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -255,29 +235,27 @@ export async function accumulateMetadata(
   const resolvers: ((value: ResolvedMetadata) => void)[] = []
   const generateMetadataResults: (Metadata | Promise<Metadata>)[] = []
 
-  // call each `generateMetadata function concurrently and stash their resolver
+  let resolvedTitle: AbsoluteTemplateString | null = null
+  let resolvedTwitterTitle: AbsoluteTemplateString | null = null
+  let resolvedOpenGraphTitle: AbsoluteTemplateString | null = null
+
+  // Loop over all metadata items again, merging synchronously any static object exports,
+  // awaiting any static promise exports, and resolving parent metadata and awaiting any generated metadata
+
+  let resolvingIndex = 0
   for (let i = 0; i < metadataItems.length; i++) {
-    const { metadataExport } = metadataItems[i]
+    const [metadataExport, staticFilesMetadata] = metadataItems[i]
+    let metadata: Metadata | null = null
     if (typeof metadataExport === 'function') {
+      // call each `generateMetadata function concurrently and stash their resolver
       generateMetadataResults.push(
         metadataExport(
           new Promise((resolve) => {
             resolvers.push(resolve)
           })
         )
       )
```

Comment:
This doesn't quite get at what I was suggesting. This goes back to having the metadata call be done serially.

What i was getting at was more like this
```
let generateMetadataResults = null;
let resolvers = null;
for (let i = 0; i < metadataItems.length; i++) {
  if (typeof metadataItem === 'function') {
    if (resolvers === null) {
      // we need to call all the functions concurrently
      resolvers = [];
      generateMetadataResults = [];
      for (let j = i; j < metadataItems.length; j++) {
        // ... fill up the two arrays now
      }
    }
    // we now always for sure have our arrays filled up and ready to use because we fill them
    // on the first occurrence of 'function'. It's basically just lazy invoking the functions
    
    // do the normal stuff for function types here
  } else {
    // do the normal stuff for object types here
  }
}